ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -41,7 +65,38 @@ let generate_file ~rendered ~destdir ~output_file =
     (fun () -> output_string out_chan rendered)
     ~finally:(fun () -> close_out out_chan)
 
+module StringSet = Set.Make (String)
+
+module TypeSet = Set.Make (struct
+  type t = ty
+
+  let compare = compare
+end)
+
+type string_record = {
+    mutable serializes: StringSet.t
+  ; mutable deserializes: StringSet.t
+}
+
+type tys_record = {mutable serials: TypeSet.t; mutable deserials: TypeSet.t}
+
 module Json = struct
+  type action = Serialize | Deserialize | Nothing
+
+  let ref_types = {serializes= StringSet.empty; deserializes= StringSet.empty}
```

Comment:
Defining data in a module seems strange to me.

Reply:
These related code has been split to the following PR which will raised after this one got merged.